server/src/main/java/org/elasticsearch/index/cache/bitset/BitsetFilterCache.java

```diff
@@ -105,7 +105,7 @@ static boolean shouldLoadRandomAccessFiltersEagerly(IndexSettings settings) {
         boolean loadFiltersEagerlySetting = settings.getValue(INDEX_LOAD_RANDOM_ACCESS_FILTERS_EAGERLY_SETTING);
         boolean isStateless = DiscoveryNode.isStateless(settings.getNodeSettings());
         if (isStateless) {
-            return DiscoveryNode.hasRole(settings.getNodeSettings(), DiscoveryNodeRole.INDEX_ROLE)
+            return DiscoveryNode.hasRole(settings.getNodeSettings(), DiscoveryNodeRole.SEARCH_ROLE)
                 && loadFiltersEagerlySetting
                 && INDEX_FAST_REFRESH_SETTING.get(settings.getSettings());
```

Comment:
This seems odd at first sight, as if we still rely on the fast refresh setting?

Reply:
I suppose the change here is not *that bad* to prevent merging this (seems similarly wrong, assuming my understanding is correct).

Reply:
I tried to keep the behavior of the `if` as it's in the code currently, but I agree that PR's final modifications seem inconsistent with its description, so if it's a bug it should be corrected. Preferably in an orthogonal PR and I rebase afterwards. But lets see what the Search team comments first.

Reply:
Yes the idea here was to only eager load when something gets searched. If we know we will never search on an indexing node then eager loading is unnecessary there :)

Reply:
Thanks @original-brownbear ! As it's orthogonal to this PR, and to avoid any confusion, I believe it's better to fix this in a separate PR. I opened https://github.com/elastic/elasticsearch/pull/114191 for our convenience, please feel free to review -- or make a brand new one if you'd like.

After that PR is merged, I will rebase this one.

Reply:
I'd be ok to merge this PR with this change as is - it seems it does not make it worse (and the other PR can then resolve it).

Reply:
Thanks @henningandersen ! I can merge after the stateless PR is also approved. If this is merged first, indeed I can rework/simplify https://github.com/elastic/elasticsearch/pull/114191 afterwards.

Reply:
Fixed in https://github.com/elastic/elasticsearch/pull/114191 .